### ISSUE TITLE
Slow loading on researcher-facing pages

### ISSUE THREAD

**Issue by mekline** [lookit/lookit-api#1241]
**TL;DR**: Reduce pageload times on views in the "Experimenter" section that lead to long loading delays

**Narrative**

A few researchers now have reported to us (and Mark and Melissa also observe) slow loading times on several views - these might be the result of queries struggling as the site grows!  This is a QOL issue for researchers and we'll explore a few approaches to fixing this.

**Details**

From Sabine Doebel (GARDEN): "When I click into 'experimenter' from the main page, it takes about 40s to get to the page that has the list of experiments I'm working on. from there, it takes another 36s to get into the experiment I'm working on. To save my experiment, it takes 1min 11 sec (my last attempt). Maybe that has something to do with the complexity of the experiment? Not sure. "

From Mark Sheskin: "I can share that I also experience delays when going into the /edit URL for a study (e.g., https://lookit.mit.edu/exp/studies/some4digitnumberhere/edit/ ) and that notable (30 second?) delay started when we implemented the changes associated with allowing previous participation in some other study to be an inclusion/exclusion criterion for a future study (maybe it is related to the time it takes to load those lists?). If this is the same delay that Sabine is having, it is worth knowing that this delay is causing some researchers significant friction! (In my case, I open the /edit and then go and do other steps in my workflow for admin review in the 30 seconds it takes to load)"

**Possible remedies**

(1) Refactoring the study edit/create pages

Already desirable for JSPsych related reasons, and may reduce load times by separating expensive many-to-many queries from the code entry pages. 

(To add: Issue # for study form updates)

(2) Examine & fix queries on these pages

Are there specific queries that are behind the long loads, and can we modify them to be less taxing? Blacklist/Whitelist many-to-many tables are suspected culprits, but delays are also seen on views that shouldn't (?) require these

(3) Expand server architecture

It may be that we're reaching the limits of the current cloud architecture to support the growing user base(!) - we can reach out to Burwood to see about increasing resources.

**Comment by sdoebel** [comment]
Not sure if this is the right place for this but a possibly related problem: still getting intermittent 503 and other errors. It's a deadly combo because the slow page loading leads one to try to make more edits in one go, which are then lost... Thanks for your help!

**Comment by becky-gilbert** [comment]
@mekline @okaycj I think this issue might be resolved via CJ's refactoring (#1257) and split of the old study details view into two separate views. Can this be closed?

**Comment by mekline** [comment]
yes!
